spark/v3.4/spark/src/main/java/org/apache/iceberg/spark/SortOrderToSpark.java

```diff
@@ -53,7 +53,7 @@ public SortOrder truncate(
       String sourceName, int id, int width, SortDirection direction, NullOrder nullOrder) {
     return Expressions.sort(
         Expressions.apply(
-            "truncate", Expressions.column(quotedName(id)), Expressions.literal(width)),
+            "truncate", Expressions.literal(width), Expressions.column(quotedName(id))),
```

Comment:
Had to switch it around so that `truncate` is resolvable. Spark did not support transforms with multiple arguments before.